protocol/drm-lease-v1.xml

```diff
@@ -0,0 +1,249 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
This protocol file shouldn't be added to the wlroots source tree. Instead we should grab it from wayland-protocols.

To ease development, it may be easier to add it in the wlroots source tree as a hack. Please isolate the change in a separate commit marked as "do not merge", so that we can drop it before merging the PR.

Reply:
I shoved the protocol in https://github.com/swaywm/wlroots/pull/2929/commits/74dda5de9c6d7ae1e48549201f8d5a99dd29ad4c